### PR TITLE
Optimize service observability plugin

### PR DESCRIPTION
**Pull request by hanbingleixue**
【Fix issue】#1314

[Modification content] Modify the way the service visibility plug-in monitors Netty connections

[Use case description] 1. No function changes, no modification required

[Self-test status] 1. Local static inspection and cleaning status; 2. Self-test passed

[Scope of Impact] 1. It has an impact on the user's use, and the configuration information in the file needs to be adjusted.

### CODE REVIEW

File: sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java [hunk 1]
```diff
@@ -46,39 +51,70 @@ public class CollectorServiceImpl implements CollectorService {
...
+    public void start() {
         try {
             this.client = ServiceManager.getService(GatewayClient.class);
         } catch (IllegalArgumentException e) {
             LOGGER.warning("GatewayClient is not found");
         }
+        if (client != null) {
+            listener = notificationInfo -> reconnectHandler();
+            NotificationManager.registry(listener, NettyNotificationType.CONNECTED.getClass());
+        }
```
Comment (lilai23) [inline sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java:68]: client == null 应该补充相应的处理

File: sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java [hunk 2]
```diff
@@ -46,39 +51,70 @@ public class CollectorServiceImpl implements CollectorService {
...
     public void sendServerInfo(ServerInfo serverInfo) {
+        if (client == null) {
+            return;
+        }
+        sendMessage(OperateType.ADD.getType(), serverInfo);
+    }
+
+    @Override
+    public void reconnectHandler() {
+        if (client == null) {
```
Comment (lilai23) [inline sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java:81]: 按照上面64行的写法client == null 永远是false

File: sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java [hunk 3]
```diff
@@ -46,39 +51,70 @@ public class CollectorServiceImpl implements CollectorService {
...
+    public void stop() {
+        if (client == null) {
+            return;
+        }
+        ServerInfo serverInfo = new ServerInfo();
+        serverInfo.setContractList(new ArrayList<>());
+        serverInfo.setConsanguinityList(new ArrayList<>());
+        sendMessage(OperateType.DELETE.getType(), serverInfo);
+        if (listener != null) {
+            NotificationManager.unRegistry(listener, NettyNotificationType.CONNECTED.getClass());
```
Comment (lilai23) [inline sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java:119]: 补充相应日志

File: sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java (hunk 2 above)
Comment (hanbingleixue) [reply sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java:81]: > 按照上面64行的写法client == null 永远是false

如果网关没开启的话client==null是为true的

File: sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java (hunk 1 above)
Comment (hanbingleixue) [reply sermant-plugins/sermant-service-visibility/service-visibility-service/src/main/java/com/huaweicloud/visibility/service/CollectorServiceImpl.java:66]: > client == null 应该补充相应的处理
没有处理逻辑，client为null时无法发送消息给backend，所以不需要注册监听，也不需要做其他处理

**Review by TangLeDaily — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**